libs/application-generic/src/graceful-shutdown/shutdown.service.ts

```diff
@@ -0,0 +1,7 @@
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class ShutdownService {
+  public readonly timeout =
+    parseInt(process.env.GRACEFUL_SHUTDOWN_TIMEOUT, 10) || 5000;
```

Comment:
Please add the new environment variable to all the [env.validators](https://github.com/novuhq/novu/blob/next/apps/api/src/config/env.validators.ts) files in services that need it. This ensures that the application won't startup without it. You will also be able to specify the `5000` default there to fully encapsulate the necessary fallback logic.